distributed/tests/test_worker.py

```diff
@@ -1455,6 +1456,31 @@ async def test_close_gracefully(c, s, a, b):
         assert_amm_transfer_story(key, b, a)
 
 
+@pytest.mark.parametrize("sync", [False, pytest.param(True, marks=[pytest.mark.slow])])
+@gen_cluster(client=True, nthreads=[("", 1)])
+async def test_close_while_executing(c, s, a, sync):
+    ev = Event()
+
+    if sync:
+
+        def f(ev):
+            ev.set()
+            sleep(2)
+
+    else:
+
+        async def f(ev):
+            await ev.set()
+            await asyncio.Future()  # Block indefinitely
+
+    f1 = c.submit(f, ev, key="f1")
+    await ev.wait()
+    task = next(task for task in asyncio.all_tasks() if "execute(f1)" in str(task))
```

Comment:
You can use https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.get_name rather than rely on the repr

Reply:
done